fb/app/models/user.rb

```diff
@@ -57,33 +46,34 @@ def friends?(user)
   end
 
   def recommended_friends
-    friends = User.all.map { |user| user if (!friends? user) && (mutual_friends(user).size > 1) && user != self }.compact
+    friends = User.all.map { |user| user if (!self.friends? user) && (mutual_friends(user).size > 1) }.compact
     if friends.size < 10
-      ((friends + unknown_users(self)) - [self]) - inverse_pending_friends
+      ((friends + unknown_users(self)) - [self]) - pending_requests
     else
-      (friends - [self]) - inverse_pending_friends
+      (friends - [self]) - pending_requests
     end
   end
 
   def confirmed_friends
-    friendships.map { |friendship| friendship.friend if friendship.confirmed }.compact
+    confirmed_friendships.map { |friendship| friendship.friend }
   end
 
   def confirmed_inverse_friends
-    inverse_friendships.map { |friendship| friendship.user if friendship.confirmed }.compact
+    confirmed_inverse_friendships.map { |friendship| friendship.user }
```

Comment:
- [ ] Similarly we can use `has_many :inverse_friends, through: :confirmed_inverse_friendships`.

Reply:
Thank you for providing the resource, it was very helpful